daemon/openrazer_daemon/hardware/device_base.py

```diff
@@ -104,13 +134,176 @@ def __init__(self, device_path, device_number, config, testing=False, additional
         self.load_methods()
 
+        # load last effects
+        for i in self.ZONES:
+            if self.zone[i]["present"]:
+                if self.config.has_section(self.storage_name):
+                    try:
+                        self.zone[i]["effect"] = self.config[self.storage_name][i + '_effect']
+                    except KeyError:
+                        self.zone[i]["effect"] = 'spectrum'
+                        pass
+
+                    try:
+                        self.zone[i]["active"] = bool(self.config[self.storage_name][i + '_active'])
+                    except KeyError:
+                        self.zone[i]["active"] = True
+                        pass
+
+                    try:
+                        self.zone[i]["brightness"] = float(self.config[self.storage_name][i + '_brightness'])
+                    except KeyError:
+                        self.zone[i]["brightness"] = 75.0
+                        pass
+
+                    try:
+                        for index, item in enumerate(self.config[self.storage_name][i + '_colors'].split(" ")):
+                            self.zone[i]["colors"][index] = int(item)
+                            if not 0 <= self.zone[i]["colors"][index] <= 255:
+                                raise ValueError('Color out of range')
+
+                        if len(self.zone[i]["colors"]) != 9:
+                            raise ValueError('There must be exactly 9 colors')
+
+                    except ValueError:
+                        # invalid colors. reinitialize
+                        self.zone[i]["colors"] = [0, 255, 0, 0, 255, 255, 0, 0, 255]
+                        self.logger.info("%s: Invalid colors; restoring to defaults.", self.__class__.__name__)
+                        pass
+
+                    except KeyError:
+                        # no colors. reinitialize
+                        self.zone[i]["colors"] = [0, 255, 0, 0, 255, 255, 0, 0, 255]
+                        self.logger.info("%s: No colors found; restoring to defaults.", self.__class__.__name__)
+                        pass
+
+                    try:
+                        self.zone[i]["speed"] = int(self.config[self.storage_name][i + '_speed'])
+                        self.zone[i]["wave_dir"] = int(self.config[self.storage_name][i + '_wave_dir'])
+
+                    except KeyError:
+                        self.zone[i]["speed"] = 1
+                        self.zone[i]["wave_dir"] = 1
+                        self.logger.info("%s: No speed/wave direction found; restoring to defaults.", self.__class__.__name__)
+                        pass
+
+                    if i == "backlight":
+                        effect_func_name = 'set' + self.zone[i]["effect"][0].upper() + self.zone[i]["effect"][1:]
+                    else:
+                        effect_func_name = 'set' + i[0].upper() + i[1:] + self.zone[i]["effect"][0].upper() + self.zone[i]["effect"][1:]
```

Comment:
Done.

**No, sorry. I will not use .capitalize() because I have a good reason not to:**
.capitalize() is NOT the same as `i[0].upper() + i[1:]`. It is more like `i[0].upper() + i[1:].lower()`. This means, as an example `breathRandom` will NOT be converted to `BreathRandom` but rather `Breathrandom`, which is not going to work. And so far as I know, there is no function to just capitalize the first letter. So I have to do this. Sorry.

Reply:
You're right about `.capitalize()` , but then please make a helper function to do that so this pattern doesn't have to exist 10 times in the code

Reply:
Hello. I tried.
```
daemon/openrazer_daemon/hardware/device_base.py:258:47: E0602: Undefined variable 'capitalize_first_char' (undefined-variable)
```
Why am I getting this error? I clearly declared the capitalize_first_char function as static under RazerDevice.

Reply:
Ooooh, figured out why. Done.